### PR TITLE
AP_Logger: split MCU from POWR into new messages, POWR log nan if voltages not available

### PR DESCRIPTION
**Pull request by IamPete1**
~~Saves lots of empty logs fields.~~

~~Quite tempted to split into multiple messages.~~

~~The only change here is that we would no longer log the safety flag, it would be the only none-zero thing.~~

Now just removes stuff that is not there rather than trying to selectively log.

### CODE REVIEW

**Comment by peterbarker** [comment]
From a recent Plane flight:
```
POWR 0.89%
```

There may be bigger fish to fry.

**Review by andyp1per — DISMISSED** [review]

**Comment by IamPete1** [comment]
The safety flag is in `STAT` on plane. We don't have that on other vehicles tho.

**Comment by IamPete1** [comment]
This now just removes the fields if they are unused.

**Comment by IamPete1** [comment]
Not a massive saving in either flash or log size, but does setup a pattern that we could use elsewhere.

**Comment by IamPete1** [comment]
Not tested IRL, but have greped the binarys to make sure the labels ect are coming out correctly.

**Comment by IamPete1** [comment]
@peterbarker This upsets the c++ parsing in the logger documentation test. I got it working as far as this regex and then gave up and added a white list.

https://github.com/ArduPilot/ardupilot/blob/15ae88db9e078c0aa37895669602e9596ac892df/Tools/autotest/common.py#L2764

We should probably decide if we actually want this before fixing the test in anycase...

**Comment by IamPete1** [comment]
Saves a bit of flash on MatekF405 where the extra items are not available. 
```
Binary Name      Text [B]         Data [B]     BSS (B)        Total Flash Change [B] (%)      Flash Free After PR (B)
---------------  ---------------  -----------  -------------  ----------------------------  -------------------------
arduplane        -108 (-0.0113%)  0 (0.0000%)  4 (+0.0031%)   -108 (-0.0113%)                                   28944
ardurover        -108 (-0.0130%)  0 (0.0000%)  -4 (-0.0031%)  -108 (-0.0130%)                                  153168
ardusub          -108 (-0.0136%)  0 (0.0000%)  -4 (-0.0031%)  -108 (-0.0136%)                                  189068
antennatracker   -108 (-0.0171%)  0 (0.0000%)  -4 (-0.0031%)  -108 (-0.0171%)                                  352016
arducopter-heli  -108 (-0.0117%)  0 (0.0000%)  4 (+0.0031%)   -108 (-0.0117%)                                   58888
arducopter       -108 (-0.0118%)  0 (0.0000%)  -4 (-0.0031%)  -108 (-0.0118%)                                   67848
blimp            -108 (-0.0166%)  0 (0.0000%)  4 (+0.0031%)   -108 (-0.0166%)                                  332948
```

File: libraries/AP_Logger/LogStructure.h [hunk 1]
```diff
@@ -327,17 +327,68 @@ struct PACKED log_Optflow {
...
+#ifdef ANALOG_VCC_5V_PIN
     float Vcc;
+#endif
+#if HAL_HAVE_SERVO_VOLTAGE
     float Vservo;
+#endif
     uint16_t flags;
     uint16_t accumulated_flags;
     uint8_t safety_and_arm;
+#if HAL_WITH_MCU_MONITORING
```
Comment (tridge) [inline libraries/AP_Logger/LogStructure.h:339]: having a separate MCU msg would likely give more flash savings and would fit our current structure

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by IamPete1** [comment]
Rebased and tested on CubeOrange, get both log messages.

**Comment by tridge** [comment]
tested on CUAV-X7, all good

**Review by tridge — APPROVED** [review]

**Pull request merged**